test/acceptance/tests/consul-dns/consul_dns_test.go

```diff
@@ -0,0 +1,79 @@
+package consuldns
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul-helm/test/acceptance/framework"
+	"github.com/hashicorp/consul-helm/test/acceptance/helpers"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const podName = "dns-pod"
+
+func TestConsulDNS(t *testing.T) {
+	env := suite.Environment()
+	context := env.DefaultContext(t)
+	releaseName := helpers.RandomName()
+
+	cluster := framework.NewHelmCluster(t, nil, context, suite.Config(), releaseName)
+	cluster.Create(t)
+
+	k8sClient := context.KubernetesClient(t)
+	contextNamespace := context.KubectlOptions().Namespace
+
+	dnsService, err := k8sClient.CoreV1().Services(contextNamespace).Get(fmt.Sprintf("%s-%s", releaseName, "consul-dns"), metav1.GetOptions{})
+	require.NoError(t, err)
+
+	dnsIP := dnsService.Spec.ClusterIP
+
+	createdDNSPod(t, releaseName, context)
+
+	testDNS(t, context, dnsIP)
+}
+
+func createdDNSPod(t *testing.T, releaseName string, context framework.TestContext) {
+	dnsPod := corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:    "dns",
+					Image:   "anubhavmishra/tiny-tools",
+					Command: []string{"dig", fmt.Sprintf("@%s-consul-dns", releaseName), "consul.service.consul"},
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyNever,
+		},
+	}
+	_, err := context.KubernetesClient(t).CoreV1().Pods(context.KubectlOptions().Namespace).Create(&dnsPod)
```

Comment:
For our other "test" containers like the bounce pod we'd used a YAML fixture. This one seems like the same use-case so I think we should be consistent. I think I like the YAML because it's easier to debug the tests ourselves.

Reply:
My initial intention was to do that as well, but the name of the `consul-dns` service is dependent on the release name. See L49. I could potentially create a YAML template for the pod which I could inject the release name into, but it felt like overkill and also isn't something that can be kubectl apply-ied. Is there a work-around?

Reply:
I just had another idea of how to simplify this such that we don't need the fixture at all.

You could create a pod and get the output of `dig` all in one command:

```shell
kubectl run  -it test-dns --image anubhavmishra/tiny-tools -- dig @iryna-consul-dns consul.service.consul
```

In code it will look like:

```go
runTestPodArgs := []string{
  "run", "-it", podName, "--image", "anubhavmishra/tiny-tools", "--", "dig", fmt.Sprintf("@%s-consul-dns", releaseName), "consul.service.consul",
}
output, err := helpers.RunKubectlAndGetOutputE(t, context.KubectlOptions(), runTestPodArgs...)
require.NoError(t, err)
// cleanup
// assertions
```